### PR TITLE
SASS-7584: Upgrade income-tax-additional-info-frontend to play 3.0

### PR DESCRIPTION
**Pull request by AishaBi**
### Description
Please include a brief summary of the change you've made and the issue that it fixes.

Add a link to the relevant story in Jira
[SASS-CHANGE-ME](ADD URL HERE)

##### Before Reviewing
- [ ]  Have you pulled the branch down?
- [ ]  Have you assigned yourself to the PR?
- [ ]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [ ]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [ ]  Have you run the smoke tests?
- [ ]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [ ]  Have you run the smoke tests?
- [ ]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

**Comment by AishaBi** [comment]
changing org.joda.time -> java.time: 1)is ZoneOffset correct or should it be ZoneID, 2) also is DateTime need to be changed to LocalDate or LocalDateTime?

File: build.sbt [hunk 1]
```diff
@@ -62,39 +62,47 @@ lazy val microservice = Project(appName, file("."))
...
-    // To resolve a bug with version 2.x.x of the scoverage plugin - https://github.com/sbt/sbt/issues/6997
-    libraryDependencySchemes ++= Seq("org.scala-lang.modules" %% "scala-xml" % VersionScheme.Always,
-    )
-  )
   .settings(PlayKeys.playDefaultPort := 10005)
   .settings(
     majorVersion := 0,
     scalaVersion := "2.13.12",
-    libraryDependencies ++= AppDependencies.compile ++ AppDependencies.test,
+    libraryDependencies ++= AppDependencies.compile ++ AppDependencies.test ++ AppDependencies.itDependencies,
```
Comment (ghousedev) [inline build.sbt:69]: AppDependencies.itDependencies is not needed here as the integration tests are no longer part of the project defined here.

File: build.sbt [hunk 2]
```diff
@@ -62,39 +62,47 @@ lazy val microservice = Project(appName, file("."))
...
   .disablePlugins(JUnitXmlReportPlugin)
-  .settings(
-    // To resolve a bug with version 2.x.x of the scoverage plugin - https://github.com/sbt/sbt/issues/6997
-    libraryDependencySchemes ++= Seq("org.scala-lang.modules" %% "scala-xml" % VersionScheme.Always,
-    )
-  )
   .settings(PlayKeys.playDefaultPort := 10005)
   .settings(
     majorVersion := 0,
     scalaVersion := "2.13.12",
```
Comment (ghousedev) [inline build.sbt:68]: Could you please add: 
`ThisBuild / majorVersion := 0
ThisBuild / scalaVersion := "2.13.12"`

to the top of the file and remove the settings for this in here and also for it.

File: build.sbt [hunk 3]
```diff
@@ -62,39 +62,47 @@ lazy val microservice = Project(appName, file("."))
...
+    Assets / pipelineStages := Seq(concat)
   )
+
+lazy val it = project
+  .enablePlugins(PlayScala)
+  .dependsOn(microservice % "test->test") // the "test->test" allows reusing test code and test dependencies
+  .settings(DefaultBuildSettings.itSettings())
+  .settings(scalaVersion := "2.13.12")
+  .settings(majorVersion := 0)
+  .settings(libraryDependencies ++= AppDependencies.test)
```
Comment (ghousedev) [inline build.sbt:106]: this line(106) is not needed

File: build.sbt [hunk 4]
```diff
@@ -62,39 +65,38 @@ lazy val microservice = Project(appName, file("."))
...
-    scalaVersion := "2.13.12",
     libraryDependencies ++= AppDependencies.compile ++ AppDependencies.test,
     TwirlKeys.templateImports ++= twirlImports,
-    Assets / pipelineStages := Seq(gzip),
+    // only required for frontends
     scalacOptions += "-Wconf:cat=unused-imports&src=html/.*:s",
-    scalacOptions += "-Wconf:src=routes/.*:s"
+    // for all services
+    scalacOptions += "-Wconf:cat=unused-imports&src=routes/.*:s"
+   //scalacOptions += "-Wconf:src=routes/.*:s"
```
Comment (ghousedev) [inline build.sbt:76]: Could you please uncomment this line, thanks

**Review by ghousedev — APPROVED** [review]

**Review by MadStu — APPROVED** [review]

**Review by MadStu — APPROVED** [review]

**Review by ghousedev — APPROVED** [review]

**Pull request merged**